test/mongodb.test.js

```diff
@@ -1072,6 +1072,111 @@ describe('mongodb connector', function () {
+    it('should create a model instance even if it already exists', function(done) {
+      Product.replaceOrCreate({name: 'newFoo'}, function(err, updatedProduct) {
+        should.not.exist(err);
+        should.not.exist(updatedProduct._id);
+        should.exist(updatedProduct.id);
+        Product.findById(updatedProduct.id, function(err, data) {
+          data.name.should.be.equal('newFoo');
+          done();
+        });
+      });
+    });
+
+    it('should replace a model instance if the passing key already exists', function(done) {
+      Product.create({name: 'foo', price: 100}, function (err, product) {
+        Product.replaceOrCreate({id: product.id, name: 'newFoo'}, function(err, updatedProduct) {
+          should.not.exist(err);
+          should.not.exist(updatedProduct._id);
+          updatedProduct.name.should.be.equal('newFoo');
+          should.exist(updatedProduct.id);
+          Product.findById(product.id, function(err, data) {
+            data.name.should.be.equal('newFoo');
+            should.not.exist(data.price);
+            done();
+          });
+        });
+      });
+    });
+
+    it('should remove extraneous properties that are not defined in the model', function(done) {
+      Product.create({name: 'foo', price: 100, 'bar': 'baz'}, function(err, product) {
+        Product.replaceOrCreate({id: product.id, name: 'newFoo'}, function(err, updatedProduct) {
+          should.not.exist(updatedProduct.bar);
+          Product.findById(product.id, function(err, data) {
+            should.not.exist(data.bar);
+            done();
+          });
+        });
+      });
+    });
+  })  
+  
+  describe('replaceAttributes', function() {
+    it('should replace the model instance if the provided key already exists', function(done) {
+      Product.create({name: 'foo', price: 100}, function(err, product) {
+        product.replaceAttributes({name: 'newFoo'}, function(err, updatedProduct) {
+          should.not.exist(err);
+          should.not.exist(updatedProduct._id);
+          updatedProduct.name.should.be.equal('newFoo');
+          should.exist(updatedProduct.id);
+          Product.findById(product.id, function(err, data) {
+            data.name.should.be.equal('newFoo');
+            should.not.exist(data.price);
+            done();
+          });
+        });
+      });
+    });
+
+    it('should remove extraneous properties that are not defined in the model', function(done) {
+      Product.create({name: 'foo', price: 100, bar: 'baz'}, function(err, product) {        
+        product.replaceAttributes({name: 'newFoo'}, function(err, updatedProduct) {
+          should.not.exist(updatedProduct.bar);
+          Product.findById(product.id, function(err, data) {
+            data.name.should.be.equal('newFoo');
+            should.not.exist(data.bar);
+            done();
+          });
+        });
+      });
+    });
+
+  });
+
+  describe('replaceAttribute', function() {
```

Comment:
If I understood your question correctly; you are asking for implementation of `replaceAttribute` in `loopback-connector-mongodb`?

But that is handled in `loopback-datasource-juggler` like the following:

```
DataAccessObject.prototype.replaceAttribute = function replaceAttribute(name, value, options, cb) {
  var data = {};
  data[name] = value;
  return this.replaceAttributes(data, options, cb);
};
```

Please correct me if I did not understand your question or I'm mistaken